jobsdb/jobsdb.go

```diff
@@ -1620,21 +1630,36 @@ func (jd *HandleT) renameDS(ds dataSetT, allowMissing bool) {
 	var sqlStatement string
 	var renamedJobStatusTable = fmt.Sprintf(`pre_drop_%s`, ds.JobStatusTable)
 	var renamedJobTable = fmt.Sprintf(`pre_drop_%s`, ds.JobTable)
+	err := jd.doInTransaction(func(tx *sql.Tx) error {
```

Comment:
renaming the table and executing the jobsdb's pre-backup handlers is performed in a single transaction.